addons/interaction/CfgVehicles.hpp

```diff
@@ -648,12 +648,11 @@ class CfgVehicles {
             class ACE_MainActions {
                 displayName = CSTRING(MainAction);
                 distance = 3;
-                position = QUOTE(_target worldToModel getPos _target);
+                position = QUOTE(_target worldToModel ASLToAGL ATLToASL getPosATL _target);
```

Comment:
How about getPosASL and just ASLToAGL?

Reply:
will check

Reply:
Why was getPos even a problem here?

Reply:
bad position under water and/or on building, I don't remember. In building `getPos` Z=0 IIRC.

Reply:
I think that's just the nature of AGL. In buildings it uses the ground under your feet so height is almost always 0. I don't understand how converting ASL to AGL makes a difference then

Reply:
`ASLToAGL` doesn't take building height into account unlike `getPos`

Reply:
I think you just want ATL then.. I guess.

Reply:
ATL didn't work under water. It really works now with AGL.